### ISSUE TITLE
GTR T5 Large Model output failure in TensorRT 8.6.1 when running on fp16 precision

### ISSUE THREAD

**Issue by kiransarv** [NVIDIA/TensorRT#3206]
## Description

When I tried to convert gtr t5 model using trtexec command in the TensorRT docker container, the model is getting converted but is giving zero outputs, With fp32 there is no issue.

## Environment

**TensorRT Version**: 8.6.1

**NVIDIA GPU**: A100

**NVIDIA Driver Version**: 525.85.12  

**CUDA Version**: 12.1

**CUDNN Version**:

Operating System:

Python Version (if applicable): 3.10

Tensorflow Version (if applicable):

PyTorch Version (if applicable): 2.0.1

Baremetal or Container (if so, version):  23.07-py3 tensorrt

## Steps To Reproduce

1. Convert gtr-t5-large model from Hugging face to onnx using optimum-cli
2. Convert the model from onnx to trt using trtexec with fp16 flag enabled.
3. Infer the text, and we will get zeroes.

As most likely the issue is with overflow LayerNorm, Tried configurations like precisionConstraints (prefer, obey), converting to onnx using latest opset but still the  output is coming as 0's and the issue persists.

**Comment by zerollzeng** [comment]
Could you please provide the onnx here? you can upload it to google drive and I can try reproduce, thanks!

**Comment by kiransarv** [comment]
Hi @zerollzeng Thanks for the reply... Here is the link to onnx model logs https://drive.google.com/drive/folders/1UwdhmhI598-nXTl9cZWg5hTcmp8Prk8C?usp=sharing

**Comment by nvluxiaoz** [comment]
@zerollzeng We had run into this issue before. From our [HuggingFace demo](https://github.com/NVIDIA/TensorRT/blob/release/9.0/demo/HuggingFace/T5/export.py#L165), which includes T5, we are still running encoder in fp32 as of ToT, and decoder in fp16 because of the severe accuracy loss in encoder fp16. If you are running an enterprise-level application, you only need to run encoder once, so you can use fp32 as a WAR. The acceleration will not be that significant compared to decoder. There are some operations in the implementation of T5 encoder (Like LayerNorm) that are significantly unfriendly to fp16. Please let me know if we want to help this customer execute where his model got overflow.
